### PR TITLE
feat: add error for fps bigger than frame timeout

### PR DESCRIPTION
**Pull request by eduardacoppo**
# What is this PR for

This PR raises an exception when frame timeout is smaller than FPS
[sc-62752]

# How I did it

# Results, How I tested

https://github.com/user-attachments/assets/c576e073-811b-4235-b865-6a647a2bdc1d

# Checklist
- [x] Assign yourself  in the PR

### CODE REVIEW

**Review by higor-melo — COMMENTED** [review]
You can also do it if the exposure time is higher than the fps period, which will also break.

File: tasks/Task.cpp [hunk 1]
```diff
@@ -906,6 +906,12 @@ void Task::acquisitionConfiguration(Arena::IDevice& device)
         ptp_sync_frame_rate->SetValue(config.frame_rate);
     }
     else {
+        double frame_timeout = _image_config.get().frame_timeout.toSeconds();
+        double seconds_per_frame = 1 / config.frame_rate;
+        if (frame_timeout < seconds_per_frame) {
+            exception(FPS_SURPASSES_FRAME_TIMEOUT);
+            return;
+        }
```
Comment (doudou) [inline tasks/Task.cpp]: This actually will let the configuration continue.

throw an exception instead.

File: tasks/Task.cpp [hunk 2]
```diff
@@ -906,6 +906,12 @@ void Task::acquisitionConfiguration(Arena::IDevice& device)
         ptp_sync_frame_rate->SetValue(config.frame_rate);
     }
     else {
+        double frame_timeout = _image_config.get().frame_timeout.toSeconds();
+        double seconds_per_frame = 1 / config.frame_rate;
+        if (frame_timeout < seconds_per_frame) {
+            throw std::runtime_error("FPS SURPASSES FRAME TIMEOUT");
+            return;
```
Comment (doudou) [inline tasks/Task.cpp]: `throw` already terminates the control flow. `return` is not needed here.

I suggest you get some reading about C++ exceptions.

File: tasks/Task.cpp [hunk 3]
```diff
@@ -906,6 +906,12 @@ void Task::acquisitionConfiguration(Arena::IDevice& device)
         ptp_sync_frame_rate->SetValue(config.frame_rate);
     }
     else {
+        double frame_timeout = _image_config.get().frame_timeout.toSeconds();
+        double seconds_per_frame = 1 / config.frame_rate;
+        if (frame_timeout < seconds_per_frame) {
```
Comment (higor-melo) [inline tasks/Task.cpp:911]:
```suggestion
        double time_length_per_frame = 1 / config.frame_rate;
        if (frame_timeout < time_length_per_frame) {
```

**Review by higor-melo — APPROVED** [review]

File: tasks/Task.cpp (hunk 3 above)
Comment (doudou) [reply tasks/Task.cpp:911]: That's nitpicking ... the time is in seconds.

File: tasks/Task.cpp [hunk 4]
```diff
@@ -906,6 +906,12 @@ void Task::acquisitionConfiguration(Arena::IDevice& device)
...
+            throw std::runtime_error("THE ACQUISITION PERIOD NEEDED BY THE CONFIGURED "
+                                     "FRAME_RATE IS BIGGER THAN THE FRAME TIMEOUT");
```
Comment (doudou) [inline tasks/Task.cpp:913]: And no need to shout ! :P

**Review by doudou — APPROVED** [review]

**Pull request merged**